tests/gen/README.md

```diff
@@ -0,0 +1,156 @@
+# Automatic Documents Generation 
+
+`AutoGenerate` and `AutoGenerateFromSDL` are a highly versatile function designed for dynamic document generation, perfect for testing and simulation purposes. 
+
+`AutoGenerateFromSDL` creates documents based on a specified schema, allowing for extensive customization of data generation. 
+
+The function generates documents adhering to a defined schema and it's configuration.
+It interprets the types and relationships within the schema to create realistic, interconnected data structures.
+
+`AutoGenerate` creates documents based on the provider collections' definitions (`[]client.CollectionDefinition`)
+
+### Demand Calculation:
+
+The functions calculate the 'demand' or the number of documents to generate based on the configuration provided.
+For related types within the schema, it intelligently adjusts the number of generated documents to maintain consistency in relationships (one-to-one, one-to-many, etc.).
+
+In the absence of explicit demands, it deduces demands from the maximum required by related types or uses a default value if no relation-based demands are present.
+
+The error will be returned if the demand for documents can not be satisfied. 
+For example, a document expects at least 10 secondary documents, but the demand for secondary documents is 5.
+
+## Configuration
+
+Both functions `AutoGenerate` and `AutoGenerateFromSDL` can be configured using options.
+
+Additionally, `AutoGenerateFromSDL` can be configured directly within the schema using annotations.
```

Comment:
todo: Similar to other comments, but I think `within the schema using ` should instead read `within the SDL doc using `